sdk/ai/ai-projects/review/ai-projects.api.md

```diff
@@ -0,0 +1,1988 @@
+## API Report File for "@azure/ai-projects"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+import { AbortSignalLike } from '@azure/abort-controller';
+import { ClientOptions } from '@azure-rest/core-client';
+import { Paged } from '@azure/core-paging';
+import { RequestParameters } from '@azure-rest/core-client';
+import { StreamableMethod } from '@azure-rest/core-client';
+import { TokenCredential } from '@azure/core-auth';
+
+// @public
+export interface AgentDeletionStatusOutput {
+    deleted: boolean;
+    id: string;
+    object: "assistant.deleted";
+}
+
+// @public
+export interface AgentEventMessage {
+    data: AgentThreadOutput | ThreadRunOutput | RunStepOutput | ThreadMessageOutput | MessageDeltaChunk | RunStepDeltaChunk | string;
```

Comment:
Is this type autogenerated? x | y | string is just string

Reply:
This is not autogenerated, it was a convenient interface.

Reply:
This is a TypeScript limitation, the language service will always simplify to the common type (e.g. `"foo" | "bar" | string` will look like `string` when the developer hovers over the type since the original typing gets thrown away)

This is why we tend to mark these as `string` and then have some kind of "known values" enum

Reply:
I pulled down the code and see it a little better now. It sounds like each of these types are objects (not strings), but the payload can also be a string. Is that accurate? Is one of the payloads actually a string? If so, that probably is reasonable. If they're all objects I would remove the string.

Generally adding values to a union type on output is a breaking change, but adding values to a union type on input is not.

For example:

```ts
interface Foo { data: Bar } => interface Foo { data: Bar | Baz }
```

Will be a breaking change for output types. There's a comment here that says "  We may add additional events over time, so we recommend handling unknown events gracefully
  in your code.**/" -> is that really true? Wouldn't that violate Azure's breaking change policy once the spec GA's?

For now, though, while in preview, it's fine to add new types to a union - while it is a breaking change for customers they should expect such changes in previews and adjust accordingly. For now though it's probably fine but it's something that'll come up on GA I think.

@xirzec @mpodwysocki @jeremymeng any thoughts on how to better design this interface in this case?